app/models/user.rb

```diff
@@ -16,6 +17,36 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    @remember_token = User.new_token
+    update remember_digest: User.digest(@remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.blank?
+
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```

Comment:
```suggestion
    update_column :remember_digest, nil
```
Phân biệt `update_column` và `update_attribute` cho chị nghe

Reply:
`update_attribute`: gọi từ object của class model, trường được update sẽ bỏ qua validates, callbacks vẫn chạy, updated_at vẫn cập nhật.
`update_column`: update trực tiếp vào DB, bỏ qua validates và callbacks, cập nhật updated_at.